### ISSUE TITLE
Integrate encountered a Lisp error: The value 2 is not of type LIST.

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1861]
Imported from SourceForge on 2024-07-05 09:27:28
Created by **[johnm693](https://sourceforge.net/u/johnm693/)** on 2014-05-28 23:58:43
Original: https://sourceforge.net/p/maxima/bugs/2738

---

    x(t):=2*cos(t);
    y(t):=2*sin(t);
    integrate((x(t)+y(t)^2)*sqrt(diff(x(t),t)^2+diff(y(t),t)^2),t);

    Maxima encountered a Lisp error:

    The value 2 is not of type LIST.

    Automatically continuing.
    To enable the Lisp debugger set *debugger-hook* to nil.
    (%i6) bug_report ();

    Please report bugs to:
        http://sourceforge.net/p/maxima/bugs
    To report a bug, you must have a Sourceforge account.
    Please include the following information with your bug report:
    -------------------------------------------------------------
    ?%build_info("5.33.0","2014-05-23 11:22:17","x86_64-unknown-linux-gnu","SBCL",
             "1.1.18")
    -------------------------------------------------------------
    The above information is also reported by the function 'build_info'.

    (%o6) ""

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 09:27:29
Created by **[villate](https://sourceforge.net/u/villate/)** on 2014-05-29 00:33:38
Original: https://sourceforge.net/p/maxima/bugs/2738/#7af5

---

- Description has changed:

Diff:

~~~~

--- old
+++ new
@@ -1,23 +1,23 @@
-x(t):=2*cos(t);
-y(t):=2*sin(t);
-integrate((x(t)+y(t)^2)*sqrt(diff(x(t),t)^2+diff(y(t),t)^2),t);
+    x(t):=2*cos(t);
+    y(t):=2*sin(t);
+    integrate((x(t)+y(t)^2)*sqrt(diff(x(t),t)^2+diff(y(t),t)^2),t);
 
-Maxima encountered a Lisp error:
+    Maxima encountered a Lisp error:
 
- The value 2 is not of type LIST.
+    The value 2 is not of type LIST.
 
-Automatically continuing.
-To enable the Lisp debugger set *debugger-hook* to nil.
-(%i6) bug_report ();
+    Automatically continuing.
+    To enable the Lisp debugger set *debugger-hook* to nil.
+    (%i6) bug_report ();
 
-Please report bugs to:
-    http://sourceforge.net/p/maxima/bugs
-To report a bug, you must have a Sourceforge account.
-Please include the following information with your bug report:
--------------------------------------------------------------
-?%build_info("5.33.0","2014-05-23 11:22:17","x86_64-unknown-linux-gnu","SBCL",
+    Please report bugs to:
+        http://sourceforge.net/p/maxima/bugs
+    To report a bug, you must have a Sourceforge account.
+    Please include the following information with your bug report:
+    -------------------------------------------------------------
+    ?%build_info("5.33.0","2014-05-23 11:22:17","x86_64-unknown-linux-gnu","SBCL",
              "1.1.18")
--------------------------------------------------------------
-The above information is also reported by the function 'build_info'.
+    -------------------------------------------------------------
+    The above information is also reported by the function 'build_info'.
 
-(%o6) ""
+    (%o6) ""

~~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 09:27:33
Created by **[villate](https://sourceforge.net/u/villate/)** on 2014-05-29 00:36:49
Original: https://sourceforge.net/p/maxima/bugs/2738/#16e7

---

Yes, I agree that there is a bug but just in case you have not noticed it, the bug does not show up if you use trigsimp:

    (%i3) integrate(trigsimp((x(t)+y(t)^2)*sqrt(diff(x(t),t)^2+diff(y(t),t)^2)),t);
                               sin(2 t)
    (%o3)               4 (t - --------) + 4 sin(t)
                                  2

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 09:27:37
Created by **[slitvinov](https://sourceforge.net/u/slitvinov/)** on 2014-05-31 09:11:17
Original: https://sourceforge.net/p/maxima/bugs/2738/#c3ba

---

The first "bad" commit 475ce5d831112

https://sourceforge.net/p/maxima/code/ci/475ce5d8311126f389830b2c8f94823724bcacc5

    Rename special variables a, b, c, d to *a*, *b*, *c*, *d* to prevent
    collisions in other files. (I'm pretty sure these are meant to be
    local to this file.)
    Test suites passes.

	Modified   src/sin.lisp

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 09:27:40
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2014-06-05 01:26:07
Original: https://sourceforge.net/p/maxima/bugs/2738/#af8f

---

Thanks for this report. Fixed by commit [84c328a5]. Closing this report.
